administrator/components/com_joomlaupdate/extract.php

```diff
@@ -0,0 +1,1635 @@
+
+			return false;
+		}
+
+		// Does the file have any data, at all?
+		if ($this->fileHeader->compressed == 0)
+		{
+			// No file data!
+			if (is_resource($outfp))
+			{
+				@fclose($outfp);
+			}
+
+			$this->runState = self::AK_STATE_DATAREAD;
+
+			return true;
+		}
+
+		// Simple compressed files are processed as a whole; we can't do chunk processing
+		$zipData = $this->fread($this->fp, $this->fileHeader->compressed);
+
+		while ($this->binStringLength($zipData) < $this->fileHeader->compressed)
+		{
+			// End of local file before reading all data?
+			if ($this->isEOF())
+			{
+				$this->setError('The archive file is corrupt or truncated');
+
+				return false;
+			}
+		}
+
+		switch ($this->fileHeader->compression)
+		{
+			case 'gzip':
+				/** @noinspection PhpComposerExtensionStubsInspection */
+				$unzipData = gzinflate($zipData);
+				break;
+
+			case 'bzip2':
+				/** @noinspection PhpComposerExtensionStubsInspection */
+				$unzipData = bzdecompress($zipData);
+				break;
+
+			default:
+				$this->setError(sprintf('Unknown compression method %s', $this->fileHeader->compression));
+
+				return false;
+				break;
+		}
+
+		unset($zipData);
+
+		// Write to the file.
+		if (is_resource($outfp))
+		{
+			@fwrite($outfp, $unzipData, $this->fileHeader->uncompressed);
+			@fclose($outfp);
+		}
+
+		unset($unzipData);
+
+		$this->runState = self::AK_STATE_DATAREAD;
+
+		return true;
+	}
+
+	/**
+	 * Set up the maximum execution time
+	 *
+	 * @return  void
+	 */
+	private function setupMaxExecTime(): void
+	{
+		$configMaxTime     = self::MAX_EXEC_TIME;
+		$bias              = self::RUNTIME_BIAS / 100;
+		$this->maxExecTime = min($this->getPhpMaxExecTime(), $configMaxTime) * $bias;
+	}
+
+	/**
+	 * Get the PHP maximum execution time.
+	 *
+	 * If it's not defined or it's zero (infinite) we use a fake value of 10 seconds.
+	 *
+	 * @return integer
+	 */
+	private function getPhpMaxExecTime(): int
+	{
+		if (!@function_exists('ini_get'))
+		{
+			return 10;
+		}
+
+		$phpMaxTime = @ini_get("maximum_execution_time");
+		$phpMaxTime = (!is_numeric($phpMaxTime) ? 10 : @intval($phpMaxTime)) ?: 10;
+
+		return max(1, $phpMaxTime);
+	}
+}
+
+// Skip over the mini-controller for testing purposes
+if (defined('_JOOMLA_UPDATE_TESTING'))
+{
+	return;
+}
+
+/**
+ * Invalidate a file in OPcache.
+ *
+ * Only applies if the file has a .php extension.
+ *
+ * @param   string  $file  The filepath to clear from OPcache
+ *
+ * @return  boolean
+ */
+function clearFileInOPCache(string $file): bool
+{
+	static $hasOpCache = null;
+
+	if (is_null($hasOpCache))
+	{
+		$hasOpCache = ini_get('opcache.enable')
+			&& function_exists('opcache_invalidate')
+			&& (!ini_get('opcache.restrict_api') || stripos(realpath($_SERVER['SCRIPT_FILENAME']), ini_get('opcache.restrict_api')) === 0);
+	}
+
+	if ($hasOpCache && (strtolower(substr($file, -4)) === '.php'))
+	{
+		return \opcache_invalidate($file, true);
```

Comment:
Qualifier is unnecessary and can be removed. Only needs to be root namespace for performance, if in a namespace file. 

```suggestion
		return opcache_invalidate($file, true);
```